### PR TITLE
Protobuf primitive definition and cleanup

### LINKED ISSUE #1415: Define New Primitives and Primitive System
### Description of the task
With the development of the new trajectory planner, genevaa drive, and new protobuf framework (#1386), we need to plan a new set of primitives to define the low-level actions that we need robots to be able to do now, and into the near future.
Some goals include: 
* Extendible to new robot capabilities, such as curve shots
* Support RobotDiagnostics and debug
* Support interesting gameplay
* Simplicity (we prefer complexity to live in AI rather than in the primitives)

### Acceptance criteria

- [ ] Plan a new set of primitives
- [ ] Develop a plan for development, integration, and testing

### Blocked By
Controller

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
* Defined new primitives in protobuf as identified in https://github.com/UBC-Thunderbots/Software/issues/1415
* Did refactoring and cleaned up unneeded code

### Testing Done
CI

### Resolved Issues
resolves https://github.com/UBC-Thunderbots/Software/issues/1415

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by MathewMacDougall** [comment]
@jonathanlew let us know when you want a review of this

**Comment by jonathanlew** [comment]
> @jonathanlew let us know when you want a review of this

Ready for review!

File: src/software/ai/intent/spinning_move_intent.h
Comment (kelvinkoon) [inline src/software/ai/intent/spinning_move_intent.h]: Nit: Whitespace between param and return.

**Review by kelvinkoon — CHANGES_REQUESTED** [review]
LGTM, nice cleaning :+1:

File: src/software/ai/hl/stp/play/kickoff_friendly_play.cpp
Comment (garethellis0) [inline src/software/ai/hl/stp/play/kickoff_friendly_play.cpp:132]: What's this change for?

File: src/software/ai/intent/chip_intent.h
Comment (garethellis0) [inline src/software/ai/intent/chip_intent.h:23]: Should the chip origin not be the ball location? I think that @MathewMacDougall and I made an explicit decision to make it the ball location when we came up with this originally, but maybe we want to consider changing that now?

File: src/software/ai/intent/spinning_move_intent_test.cpp
Comment (garethellis0) [inline src/software/ai/intent/spinning_move_intent_test.cpp]: Should we have the space in the name now?

File: src/software/proto/message_translation/BUILD
Comment (garethellis0) [inline src/software/proto/message_translation/BUILD:31]: Perhaps `convert_geom_to_proto` might be a bit more descriptive? I'm not sure the `tbots_` prefix really adds any value. Similarly for the file names and the test below.

File: src/software/proto/sensor_msg.proto
Comment (garethellis0) [inline src/software/proto/sensor_msg.proto]: Hmm, so we don't have the `Tbots` prefix on any of our other message types. Any particular reason we think we should keep it on `TbotsRobotStatusMsg`

**Review by garethellis0 — CHANGES_REQUESTED** [review]
Lots of great work here! Really happy to see the new proto structure getting some use :smile:

File: src/software/proto/sensor_msg.proto
Comment (jonathanlew) [reply src/software/proto/sensor_msg.proto]: I think we should differentiate between messages we define and messages that ssl define

File: src/software/proto/message_translation/BUILD
Comment (jonathanlew) [reply src/software/proto/message_translation/BUILD:31]: I thought the folder name would make it clear, i.e. message_translation for tbots_geometry protos

File: src/software/ai/intent/chip_intent.h
Comment (jonathanlew) [reply src/software/ai/intent/chip_intent.h:23]: this was a typo from a while back, everywhere in the codebase we explicitly set a chip_origin. I think that it makes sense if we want to intercept the ball for a chip

File: src/software/ai/hl/stp/play/kickoff_friendly_play.cpp
Comment (jonathanlew) [reply src/software/ai/hl/stp/play/kickoff_friendly_play.cpp:132]: I've removed the chip_distance if we define the chip_target, since chip_distance = chip_target-chip_origin

File: src/shared/proto/primitive.proto
Comment (jonathanlew) [reply src/shared/proto/primitive.proto:34]: no, we can't do that without a lot more work in integration. I don't think a 5000 line PR is a good idea. This PR is just for defining these protos. I'll make more issues soon:tm:

File: src/firmware/app/primitives/spinning_move_primitive.c
Comment (jonathanlew) [reply src/firmware/app/primitives/spinning_move_primitive.c:66]: this needs to be done to all primitives so making an issue: https://github.com/UBC-Thunderbots/Software/issues/1631

File: src/firmware/app/primitives/spinning_move_primitive.c
Comment (jonathanlew) [reply src/firmware/app/primitives/spinning_move_primitive.c]: changed

File: src/shared/proto/vision.proto
Comment (jonathanlew) [reply src/shared/proto/vision.proto:16]: I don't think it's worth changing it since being explicit will help avoid confusion. In any case, I don't think it's necessary for it to be done in this PR

File: src/software/ai/intent/spinning_move_intent.h
Comment (jonathanlew) [reply src/software/ai/intent/spinning_move_intent.h]: fixed

**Review by kelvinkoon — COMMENTED** [review]
LGTM besides outstanding comments.

File: src/software/proto/sensor_msg.proto
Comment (garethellis0) [reply src/software/proto/sensor_msg.proto]: So we might want to have the `Tbots` prefix, or a `tbots` namespace, but either way, I think that's outside of the scope of this PR. I think we should file a ticket to update it in the future and change this to be consistent with what is in the code now.

File: src/software/proto/message_translation/BUILD
Comment (garethellis0) [reply src/software/proto/message_translation/BUILD:31]: Ah yeah. I agree. This is fine with me.

File: src/software/ai/intent/spinning_move_intent_test.cpp
Comment (jonathanlew) [reply src/software/ai/intent/spinning_move_intent_test.cpp]: fixed

File: src/shared/proto/geometry.proto
Comment (akhilveeraghanta) [inline src/shared/proto/geometry.proto:6]: Not sure if it makes sense for a point to have a unit? The unit is interpreted by the thing using Point. 

But don't feel strongly about this, because we generally only use Points for things on the field which is in meters.

File: src/shared/proto/geometry.proto
Comment (akhilveeraghanta) [inline src/shared/proto/geometry.proto]: copy pasta err

File: src/software/ai/intent/intent_visitor.h
Comment (akhilveeraghanta) [inline src/software/ai/intent/intent_visitor.h:33]: lets update the RFC to reflect these names so we still track 1:1

**Review by akhilveeraghanta — COMMENTED** [review]
Overall LGTM, just a few comments

File: src/shared/proto/geometry.proto
Comment (jonathanlew) [reply src/shared/proto/geometry.proto:6]: hm how can we make this consistent with Angle, which is in radians?

File: src/shared/proto/geometry.proto
Comment (jonathanlew) [reply src/shared/proto/geometry.proto]: nice catch

**Review by LiCody — COMMENTED** [review]
LGTM pending other reviewers comments

File: src/software/proto/sensor_msg.proto
Comment (jonathanlew) [reply src/software/proto/sensor_msg.proto]: I'm not sure exactly what you want to see, so I'm not sure what to put in this ticket

File: src/shared/proto/geometry.proto
Comment (akhilveeraghanta) [reply src/shared/proto/geometry.proto:6]: yeah thats a fair point, don't want to block this PR on this, we can revisit it if it becomes a problem later

**Review by akhilveeraghanta — DISMISSED** [review]
🧼

File: src/software/proto/sensor_msg.proto
Comment (garethellis0) [reply src/software/proto/sensor_msg.proto]: Sorry, I think we should name this `RobotStatusMsg` for now to be consistent with other messages (ie. no prefix) and file a ticket to consider adding the `Tbots` prefix or putting all messages in a `tbots` namespace.

File: src/software/primitive/spinning_move_primitive.h
Comment (kelvinkoon) [inline src/software/primitive/spinning_move_primitive.h]: Nit: Whitespace b/w param and return.

File: src/software/primitive/spinning_move_primitive.h
Comment (kelvinkoon) [inline src/software/primitive/spinning_move_primitive.h]: Nit: Whitespace b/w param and return.

**Review by kelvinkoon — CHANGES_REQUESTED** [review]
Small nits, LGTM otherwise.

File: src/software/primitive/spinning_move_primitive.h
Comment (jonathanlew) [reply src/software/primitive/spinning_move_primitive.h]: done

File: src/software/proto/sensor_msg.proto
Comment (jonathanlew) [reply src/software/proto/sensor_msg.proto]: ah I see, done

**Review by kelvinkoon — APPROVED** [review]
LGTM! :broom:

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:

**Pull request merged**
